Automattic/woocommerce-payments
# Add digital wallet buttons previews to settings page

Fixes https://github.com/Automattic/woocommerce-payments/issues/1808

#### Changes proposed in this Pull Request
Add a preview for Google Pay/ Apple Pay in the settings section. This button preview uses Stripe's [Payment request button](https://stripe.com/docs/stripe-js/elements/payment-request-button) so it looks exactly the same as the checkout page. This PR includes the following changes:
- Create 2 new components to render the preview button 
- This preview button uses Stripe's payment request button with some pre-defined values (currency, locale, amount). 
- This preview button does not popup the actual payment window.









#### Notes
1. On Safari, clicking on the different options sometimes do not re-render the button. We don't have this problem on Chrome. 
2. Google Pay does not have "Book" option and will fall back to the "default" icon.

#### Testing instructions

1. Make sure you are using `https://`. WIth `_wcpay_feature_grouped_settings` enabled. 
2. Open your Chrome browser and go to `wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments&method=payment_request`
3. You should see a button at the bottom: 
![image](https://user-images.githubusercontent.com/572862/126710671-fe90f45d-dd24-42df-a22a-4ce3b04b1a26.png)
4. Choosing different options will update the button preview.
5. Open up Safari in private viewing mode. Repeat the same and you should see the button changes. The buttons should look like these:
Apple Pay:
![image](https://user-images.githubusercontent.com/572862/127381831-c212b3cf-6013-4ec0-b293-476384653254.png)
No available methods:
![image](https://user-images.githubusercontent.com/572862/127381887-3fdb30ac-287a-4531-9252-4a0f2dc64519.png)
Google Pay:
![image](https://user-images.githubusercontent.com/572862/127381998-f8008c86-311a-4364-a048-27e752bd7bef.png)




-------------------
